File: BlackJack/BlackJack_csharp/BlackJack/BlackJack/model/GameFacade.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BlackJack.model
{
	class GameFacade
	{
		Dealer m_dealer;
		Player m_player;

		public GameFacade()
		{
			m_dealer = new Dealer(new rules.ConcreteRules.RuleFactory());
			m_player = new Player();
		}

		public void StartNewRound()
		{
			m_dealer.StartNewRound(m_player);
		}

		public void Hit()
		{
			m_dealer.HitPlayer(m_player);
		}

		public void Stand()
		{
			m_dealer.Stand();
		}

		public IEnumerable<Card> GetDealerHand()
		{
			return m_dealer.GetHand();
		}

		public IEnumerable<Card> GetPlayerHand()
		{
			return m_player.GetHand();
		}

		public int GetDealerScore()
		{
			return m_dealer.GetScoreOfHand();
		}

		public int GetPlayerScore()
		{
			return m_player.GetScoreOfHand();
		}

		public bool IsPlayerWinner()
		{
			return m_dealer.IsPlayerWinner(m_player);
		}

		public bool IsGameOver()
		{
			return m_dealer.IsGameOver();
		}
	}
}

File: BlackJack/BlackJack_java/model/GameFacade.java
package model;

public class GameFacade {
  Dealer m_dealer;
  Player m_player;
  
  public GameFacade() {
    model.rules.RuleFactory rules = new model.rules.RuleFactory();
    m_dealer = new Dealer(rules);
    m_player = new Player();
  }

  public void start() {
    m_dealer.startGame(m_player);
  }
  
  public void hit() {
    m_dealer.hit(m_player);
  }
  
  public void stand() {
    m_dealer.stand(m_player);
  }
  
  public int getDealerScore() {
    return m_dealer.getScore();
  }
  
  public int getPlayerScore() {
    return m_player.getScore();
  }
  
  public boolean isGameOver() {
    return m_dealer.isGameOver();
  }
  
  public boolean isDealerWinner() {
    return m_dealer.isDealerWinner(m_player);
  }
  
  public Iterable<Card> getDealerHand() {
    return m_dealer.getHand();
  }
  
  public Iterable<Card> getPlayerHand() {
    return m_player.getHand();
  }
}